pkg/extensions/sync/remote.go

```diff
@@ -50,13 +51,33 @@ func (registry *RemoteRegistry) GetContext() *types.SystemContext {
 func (registry *RemoteRegistry) GetRepositories(ctx context.Context) ([]string, error) {
 	var catalog catalog
 
-	_, _, _, err := registry.client.MakeGetRequest(ctx, &catalog, "application/json", //nolint: dogsled
+	_, header, _, err := registry.client.MakeGetRequest(ctx, &catalog, "application/json", "", //nolint: dogsled
 		constants.RoutePrefix, constants.ExtCatalogPrefix)
 	if err != nil {
 		return []string{}, err
 	}
 
-	return catalog.Repositories, nil
+	repos := catalog.Repositories
+
+	link := header.Get("Link")
```

Comment:
So these repos return the "Link" even if we don't specify the query parameter?

Reply:
According to the specification: https://github.com/opencontainers/distribution-spec/blob/v1.1.0/spec.md#content-discovery

> When n is zero, this endpoint MUST return an empty list, and MUST NOT include a Link header.

Since we are not setting it to 0, its probably not an issue. However it also states further down that:

> A Link header MUST be included in the response when the descriptor list cannot be returned in a single manifest. Each response is an image index with different descriptors in the manifests field. The Link header MUST be set according to [RFC5988](https://www.rfc-editor.org/rfc/rfc5988.html) with the Relation Type rel="next".

So technically if there is only 1 page, it is not required to be in the response if I get it right. But according to https://pkg.go.dev/net/http#Header.Get an empty string is returned when there is no value associated with the requested header - or it does not exist.

Reply:
some registry might have a limit hardcoded like https://github.com/distribution/distribution/tree/main/registry and it will return links to get next page.